sdk/python/kfp/components/executor.py

```diff
@@ -53,22 +53,29 @@ def _make_output_artifact(cls, runtime_artifact: Dict):
     import os
     artifact = create_runtime_artifact(runtime_artifact)
     os.makedirs(os.path.dirname(artifact.path), exist_ok=True)
-    return  artifact
+    return artifact
 
   def _get_input_artifact(self, name: str):
     return self._input_artifacts.get(name)
 
   def _get_output_artifact(self, name: str):
     return self._output_artifacts.get(name)
 
-  def _get_input_parameter_value(self, parameter_name: str):
+  def _get_input_parameter_value(self, parameter_name: str, parameter_type: Any):
     parameter = self._input.get('inputs', {}).get('parameters',
                                                   {}).get(parameter_name, None)
     if parameter is None:
       return None
 
     if parameter.get('stringValue'):
-      return parameter['stringValue']
+      if parameter_type == str:
+        return parameter['stringValue']
+      elif parameter_type == bool:
+        # Use `.lower()` so it can also handle 'True' and 'False' (resulted from
```

Comment:
Not sure I understand this part. When does this str(True) get used?

Reply:
In `lightweight_python_functions_v2_pipeline.py`, I have an example and a comment. This adds a bit tolerance to user code, it works regardless whether they use `str()` or `json.dumps()` on boolean values -- cause I think users might not think of the need to use `json.dumps` on a boolean value.
Another option could be using `distutils.util.strtobool()`, which accepts more strings like "yes", "no", "on", "off", etc. which I think isn't necessary.